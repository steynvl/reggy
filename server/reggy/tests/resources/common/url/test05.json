{
  "type": "URL",
  "information": {
    "schemes": "http;https;www.",
    "portNumbers": "No port number",
    "specOptionalPortNumbers": "",
    "specRequiredPortNumbers": "",
    "username": "Allow any user name",
    "specUserNames": "",
    "password": "Optional password",
    "domainName": "Allow any domain name",
    "specDomainNames": "",
    "specificTld": "",
    "subdomainOnSpecDomain": "",
    "folders": "No folders",
    "minFolderDepth": "",
    "maxFolderDepth": "",
    "specFoldersOnly": "",
    "fileNames": "No file names",
    "specExtensions": "",
    "specFileNames": "",
    "optionalFileNames": true,
    "parameters": "Allow any parameters",
    "specParameters": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
