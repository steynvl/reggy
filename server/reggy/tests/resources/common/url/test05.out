Java
Pattern regex = Pattern.compile("(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?");
(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?
Perl
my $regex = /(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?/;
(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?
POSIX
(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?
(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?
Python
regex = re.compile(r'(?:https?//|www\.)(?:[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&\'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?')
(?:https?//|www\.)(?:[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&\'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&\'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?
JavaScript
const regex = /(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?/;
(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?
PHP
$regex = '/(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?/';
(?:https?\/\/|www\.)(?:[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+.\/=?`{|}~^-]+)?@(?:[\da-z-]++\.)+[a-z]{2,63}+\/?\/?(?:\?[]\d!"#$%&'()*+,.\/:;<=>?@\[\_`a-z{|}~^-]*)?
Golang
regex, _ := regexp.Compile("(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]+\\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?")
(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]+\\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?
Rust
let re = Regex::new(r"(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?").unwrap();
(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?
C#
var re = new Regex(@"(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?");
(?:https?//|www\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]+?(?::[\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\da-z-]+\.)+[a-z]{2,63}/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\_`a-z{|}~^-]*)?
Scala
val re = "(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?".r
(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?
Kotlin
val regex = Regex("(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?")
(?:https?//|www\\.)(?:[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]+?(?::[\\w!#$%&'*+./=?`{|}~^-]+)?@(?:[\\da-z-]++\\.)+[a-z]{2,63}+/?/?(?:\?[]\d!"#$%&'()*+,./:;<=>?@\[\\_`a-z{|}~^-]*)?